src/validation/__tests__/KnownOperationTypesRules-test.ts

```diff
@@ -0,0 +1,52 @@
+import { describe, it } from 'mocha';
+
+import { KnownOperationTypesRule } from '../rules/KnownOperationTypesRule';
+
+import { expectValidationErrors } from './harness';
+
+function expectErrors(queryStr: string) {
+  return expectValidationErrors(KnownOperationTypesRule, queryStr);
+}
+
+function expectValid(queryStr: string) {
+  expectErrors(queryStr).toDeepEqual([]);
+}
+
+describe('Validate: Known operation types', () => {
+  it('one known operation', () => {
+    expectValid(`
+      {
+        field
+      }
+    `);
+  });
+
+  it('one unknown operation', () => {
+    expectErrors(`
+      mutation {
+        field
+      }
+    `).toDeepEqual([
+      {
+        message: 'The mutation operation is not supported by the schema.',
+        locations: [{ line: 2, column: 7 }],
+      },
+    ]);
+  });
+
+  it('mixture of known and unknown operations', () => {
+    expectErrors(`
+      query {
+        field
+      }
+      mutation {
+        field
+      }
```

Comment:
nit: in the case of dummy fields, you can use single line notation for each operation.
```
query { field }
mutation { field }
```

Reply:
This style was taken from existing rules.

Reply:
ah, likely those files have other selection sets with more than one field and better to be consistent across the file.

Reply:
https://github.com/graphql/graphql-js/commit/36163fbffde623cac962ab7898c26ec276ae2a53